api/v1beta1/common_types.go

```diff
@@ -126,6 +126,10 @@ const (
 	// any changes to the actual object because it is a dry run) and the topology controller
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
+
+	// ReplicasManagedByAutoscalerAnnotation is an annotation that indicates an external autoscaler controls replica count.
+	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
```

Comment:
Sorry not super familiar with MachinePools. Can we expand a bit what exactly is managed by the autoscaler?

Does the autoscaler only set the replica field or does it also create/delete the actual Machines and then updates the replica field?

Reply:
Follow-up. The MachinePool.spec.replicas field would be essentially ignored? Or also updated by the autoscale so it matches status?

Reply:
> Follow-up. The MachinePool.spec.replicas field would be essentially ignored? Or also updated by the autoscaler so it matches status?

Not sure. Maybe open for discussion. I'd personally like to see it updated to match the actual just so it makes things less confusing when debugging issues.

Reply:
This annotation is meant to communicate the following:

- The underlying set of machines in this pool is under the active management of an external authority (e.g., autoscaler, or spot vm mechanism)
- Cluster API should *not* attempt to enforce its idea of replica desired state
- Cluster API replica state for MachinePool is read only from a user configuration point of view
- Essentially, Cluster API should simply, periodically ensure that it has an updated replica count that matches the external authority. The only mechanism for updating is by programatically polling the external authority and taking that external authority's notion of replica count as the correct one, and updating Cluster API resources if they disagree

Does that help?

Reply:
Yes!

And with "Cluster AP" in this context we mean the infra provider implementing the corresponding InfraMachinePool?